src/fat_beagle.cpp

```diff
@@ -489,43 +501,49 @@ PhyloGradient FatBeagle::Gradient(const UnrootedTree &in_tree) const {
       BranchGradientInternals(tree.Topology(), tree.BranchLengths(), dQ);
 
   GradientMap gradient;
-
-  // Calculate substitution model parameter gradient, if needed.
-  if (phylo_model_->GetSubstitutionModel()->GetRates().size() > 0) {
-    FatBeagle *mutable_this = const_cast<FatBeagle *>(this);
-    gradient["substitution_model"] = SubstitutionModelGradient<UnrootedTree>(
-        FatBeagle::StaticUnrootedLogLikelihood, mutable_this, in_tree);
+  // Calculate Substitution Model Gradients.
+  if (PhyloFlags::IsFlagSet(flags, PhyloGradientFlagOptions::substitution_model_)) {
+    if (phylo_model_->GetSubstitutionModel()->GetRates().size() > 0) {
+      auto [rates_grad, freqs_grad] = SubstitutionModelGradient<UnrootedTree>(
+          FatBeagle::StaticUnrootedLogLikelihood, this, in_tree);
+      auto model_grad = std::vector<double>();
+      model_grad.insert(model_grad.end(), rates_grad.begin(), rates_grad.end());
+      model_grad.insert(model_grad.end(), freqs_grad.begin(), freqs_grad.end());
+      gradient[PhyloGradientMapkeys::substitution_model_()] = model_grad;
```

Comment:
For discussion: help me understand why this is a method/function.

Reply:
GetKey, please

Reply:
Changed to just bracket accessor.

Reply:
Lovely!